ultra/ultra/env/ultra_env.py

```diff
@@ -148,41 +140,43 @@ def reset(self):
 
-    def get_scenarios(self, scenario_info: Tuple[str, str]) -> List[str]:
-        """Finds all of the scenarios from the given scenario information
-
-        To obtain the path of scenarios:
-            1. Get the filename patterns of the train and test scenarios from
-               ultra/config.yaml from the given task and level.
-            2. Change the filename patterns from relative to absolute path
-            3. Depending on the evaluation mode, we can provide glob.glob with
-               the appropriate filename pattern to return a list of scenario dirs
+    @staticmethod
+    def get_scenarios_from_scenario_info(
+        scenario_info: Union[Tuple[str, str], Sequence[str]], eval_mode: bool
+    ) -> List[str]:
+        """Finds all scenarios from a given (task, level) tuple, or sequence of scenario
+        directories.
 
         Args:
-            scenario_info Tuple[str, str]: The first index represents
-                task id and second index represents task's level
+            scenario_info (Union[Tuple[str, str], Sequence[str]]): Either a tuple of
+                two strings (task, level) that describe the scenarios of the specific
+                task and level, or a sequence of scenario directory strings.
 
         Returns:
-            List[str]: Absolute path of scenarios
+            List[str]: A list of scenario directory strings.
         """
-        task_id, task_level = scenario_info[0], scenario_info[1]
-
-        base_dir = os.path.join(os.path.dirname(__file__), "../")
-        config_path = os.path.join(base_dir, _CONFIG_FILE)
-
-        with open(config_path, "r") as task_file:
-            tasks = yaml.safe_load(task_file)["tasks"]
-            scenario_paths = tasks[f"task{task_id}"][task_level]
-
-        scenario_paths["train"] = os.path.join(base_dir, scenario_paths["train"])
-        scenario_paths["test"] = os.path.join(base_dir, scenario_paths["test"])
-
-        if not self.eval_mode:
-            _scenarios = glob.glob(f"{scenario_paths['train']}")
-        else:
-            _scenarios = glob.glob(f"{scenario_paths['test']}")
-
-        return _scenarios
+        try:
+            # Attempt to load scenarios from a (task, level) tuple.
+            task, level = scenario_info
+            base_dir = os.path.join(os.path.dirname(__file__), "../")
+            task_config_path = os.path.join(base_dir, _CONFIG_FILE)
+
+            with open(task_config_path, "r") as tasks_file:
+                tasks = yaml.safe_load(tasks_file)["tasks"]
+            scenario_paths = tasks[f"task{task}"][level]
+
+            scenario_paths["train"] = os.path.join(base_dir, scenario_paths["train"])
+            scenario_paths["test"] = os.path.join(base_dir, scenario_paths["test"])
+
+            if not eval_mode:
+                scenarios = glob.glob(f"{scenario_paths['train']}")
+            else:
+                scenarios = glob.glob(f"{scenario_paths['test']}")
+        except Exception:
```

Comment:
I believe we should identify the appropriate exception type (in the case it would an KeyError exception)

Reply:
Yes, a KeyError could occur in line 166, but also a ValueError could occur in line 160 if there are not exactly 2 elements in the `scenario_info` sequence. Are you thinking we should specify the specific exceptions in case we get an unexpected exception?

Reply:
Yeah, something like that. I think there are two cases here, one is that there can be a ValueError at line 160, however if there are exactly **2** scenario directories passed in, then we might also get a KeyError as well.  We can have something like this
on line 175

```python
except (ValueError, KeyError) as e:
```